### ISSUE TITLE
[cli] register main schemas using dataValidator.addSchema(<service>Schema)

### ISSUE THREAD

**Issue by miguelrk** [feathersjs/feathers#2993]
**TLDR:** CLI should be udpated to register main schemas using `dataValidator.addSchema(<service>Schema)` to be able to use `Type.Ref(<service>Schema)`

Relevant [thread](https://discord.com/channels/509848480760725514/930352418179391528/1065065099750023188)

### Steps to reproduce

I'm getting the same Error `MissingRefError: can't resolve reference User from id Message` when using user: Type.Ref(userSchema) for the following:
```ts
const userSchema = Type.Object({
    // ...
  }, { $id: 'User', additionalProperties: false }
)
const messageSchema = Type.Object({
    userId: Type.String(),
    user: Type.Ref(userSchema)
  }, { $id: 'Message', additionalProperties: false }
)
```

### Expected behavior

Main schemas (and all others: data, patch and query) should be registered using ajv's `addSchema()` 

### Actual behavior

I have tried the following for user instead of Type.Ref(userSchema) to no success:
```ts
// throws "Error: reference "User" resolves to more than one schema
user: userSchema
// throws "MissingRefError"
user: Type.Ref(userSchema, { $id: 'User' })
// does not work
user: Type.Ref(userSchema, { $schema: 'User' })
```

### System configuration

```
"@feathersjs/adapter-commons": "^5.0.0-pre.35",
"@feathersjs/authentication": "^5.0.0-pre.35",
"@feathersjs/authentication-client": "^5.0.0-pre.35",
"@feathersjs/authentication-oauth": "^5.0.0-pre.35",
"@feathersjs/configuration": "^5.0.0-pre.35",
"@feathersjs/errors": "^5.0.0-pre.35",
"@feathersjs/express": "^5.0.0-pre.35",
"@feathersjs/feathers": "^5.0.0-pre.35",
"@feathersjs/mongodb": "^5.0.0-pre.35",
"@feathersjs/schema": "^5.0.0-pre.35",
"@feathersjs/socketio": "^5.0.0-pre.35",
"@feathersjs/transport-commons": "^5.0.0-pre.35",
"@feathersjs/typebox": "^5.0.0-pre.35",
```

**Comment by miguelrk** [comment]
For reference, adding `dataValidator.addSchema(<service>Schema)` right after declaring `<service>Schema` gets it working. For example:

```ts
// user.schema.ts
const userSchema = Type.Object({
    // ...
  }, { $id: 'User', additionalProperties: false }
)
dataValidator.addSchema(userSchema) // ADD THIS LINE HERE

// message.schema.ts
import { userSchema } from "../users/user.schema"

const messageSchema = Type.Object({
    userId: Type.String(),
    user: Type.Ref(userSchema)
  }, { $id: 'Message', additionalProperties: false }
)
```
